src/components/MyButton.vue

```diff
@@ -1,37 +1,41 @@
 <template>
-  <button class="button is-primary" @click="onClick">
-    <!-- @slot default inner button content -->
-    <slot></slot>
+  <button
+    class="my-button"
+    :disabled="disabled"
+    @click="onClick"
+  >
+    <slot />
   </button>
 </template>
 
 <script lang="ts">
 import { defineComponent } from 'vue';
 
 export default defineComponent({
-  name: 'my-button',
-
-  methods: {
-    onClick() {
-      /**
-       * Click event
-       *
-       * @event click
-       */
-      this.$emit('click');
-    },
+  name: 'MyButton',
+  props: {
+    onClick: { type: Function as ()=> void, required: true },
```

Comment:
Events are how we communicate to parent components in Vue. Please have a look at https://v3.vuejs.org/guide/component-custom-events.html#event-names